tests/integration/modules/archive.py

```diff
@@ -141,7 +141,7 @@ def test_tar_unpack(self):
 
         # Test extract archive
         ret = self.run_function('archive.tar', ['-xvf', self.arch], dest=self.dst)
-        self.assertTrue(isinstance(ret, list))
+        self.assertTrue(isinstance(ret, list), str(ret))
```

Comment:
When this check fails, it is hard to tell what exactly went wrong. This change makes test generate output like this:

```
======================================================================
FAIL: test_cmd_zip (integration.modules.archive.ArchiveTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "salt/tests/integration/modules/archive.py", line 188, in test_cmd_zip
    self.assertTrue(isinstance(ret, list), str(ret))
AssertionError: The minion function caused an exception: Traceback (most recent call last):
  File "salt/salt/minion.py", line 1479, in _thread_return
    return_data = executor.execute()
  File "salt/salt/executors/direct_call.py", line 28, in execute
    return self.func(*self.args, **self.kwargs)
  File "<string>", line 2, in cmd_zip
  File "salt/salt/utils/decorators/__init__.py", line 184, in wrapped
    return function(*args, **kwargs)
  File "salt/salt/modules/archive.py", line 653, in cmd_zip
    cmd.extend(_expand_sources(sources))
  File "salt/salt/modules/archive.py", line 422, in _expand_sources
    sources = path
UnboundLocalError: local variable 'path' referenced before assignment
```

and the problem becomes easier to debug.